data-prepper-plugins/aggregate-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/aggregate/actions/AppendAggregateActionTest.java

```diff
@@ -0,0 +1,289 @@
+        eventMaps.add(firstEventMap);
+        events.add(buildEventFromMap(firstEventMap));
+
+        // { "secondString": "secondEventString", "secondArray": [4, 5, 6], "secondNumber": 2, "matchingNumber": 11, "matchingNumberEqual": 38947, "matchingStringEqual": "equalString", "matchingNumberArray": [23,24,25], "matchingNumberArrayEqual": [20,21,22], "matchingString": "StringFromSecondEvent", "matchingStringArray": ["String3", "String4"], "matchingDeepArray": [[30,31,32]]}
+        final Map<String, Object> secondEventMap = new HashMap<>();
+        secondEventMap.put("secondString", "secondEventString");
+        secondEventMap.put("secondArray", Arrays.asList(4, 5, 6));
+        secondEventMap.put("secondNumber", 2);
+        secondEventMap.put("matchingNumber", 11);
+        secondEventMap.put("matchingNumberEqual", 38947);
+        secondEventMap.put("matchingNumberArray", Arrays.asList(23, 24, 25));
+        secondEventMap.put("matchingNumberArrayEqual", Arrays.asList(20, 21, 22));
+        secondEventMap.put("matchingString", "StringFromSecondEvent");
+        secondEventMap.put("matchingStringEqual", "equalString");
+        secondEventMap.put("matchingStringArray", Arrays.asList("String3", "String4"));
+        secondEventMap.put("matchingDeepArray", Arrays.asList(Arrays.asList(30, 31, 32)));
+        eventMaps.add(secondEventMap);
+        events.add(buildEventFromMap(secondEventMap));
+    }
+
+    private Event buildEventFromMap(final Map<String, Object> eventMap) {
+        return JacksonEvent.builder().withEventType("event").withData(eventMap).build();
+    }
+
+    private AggregateAction createObjectUnderTest(AppendAggregateActionConfig config) {
+        return new AppendAggregateAction(config);
+    }
+
+    @Test
+    void handleEvent_with_empty_group_state_should_return_correct_AggregateResponse_and_add_event_to_groupState() throws NoSuchFieldException, IllegalAccessException {
+        AppendAggregateActionConfig appendAggregateActionConfig = new AppendAggregateActionConfig();
+        final List<String> testKeysToAppend = new ArrayList<>();
+        testKeysToAppend.add(UUID.randomUUID().toString());
+        setField(AppendAggregateActionConfig.class, appendAggregateActionConfig, "keysToAppend", testKeysToAppend);
+        appendAggregateAction = createObjectUnderTest(appendAggregateActionConfig);
+
+        final AggregateActionInput aggregateActionInput = new AggregateActionTestUtils.TestAggregateActionInput(Collections.emptyMap());
+
+        final AggregateActionResponse aggregateActionResponse = appendAggregateAction.handleEvent(events.get(0), aggregateActionInput);
+
+        assertThat(aggregateActionResponse.getEvent(), equalTo(null));
+        assertThat(aggregateActionInput.getGroupState(), equalTo(events.get(0).toMap()));
+    }
+
+    private void consumeEvent(Event event, GroupState groupState, List<String> keysToAppend) {
```

Comment:
I'm not quite sure about this approach. Can you explain some?

Reply:
I did not fully understood your question. Is the question about the approach on testing or is it about the approach on the way events are being aggregated?

If the question is about the approach on testing then:
 - I was following test cases from [PutAllAggregation](https://github.com/opensearch-project/data-prepper/blob/5ca2ca6f3b026180fdf0c0365dbd13591dac6a01/data-prepper-plugins/aggregate-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/aggregate/actions/PutAllAggregateActionTest.java#L76). Since the aggregation logic is more than few lines I thought of putting that in a method. This method is just a copy of what I am using [here](https://github.com/opensearch-project/data-prepper/blob/54fa885386df7305e2534df8211c6a02c89c0825/data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/actions/AppendAggregateAction.java#L58)

If the question is about the way events are being aggregated:
- I have documented it [here](https://github.com/opensearch-project/data-prepper/tree/54fa885386df7305e2534df8211c6a02c89c0825/data-prepper-plugins/aggregate-processor#append)

Reply:
There shouldn't be a need to duplicate the `consumeEvent` method here. Other than that all the tests look good.